tracer/test/Datadog.Trace.Tests/LambdaHandlerTests.cs

```diff
@@ -64,17 +65,30 @@ public void LambdaHandlerCanHandleClosedGenericTypes(string handlerVariable, str
+        [InlineData("Datadog.Trace.Tests::Datadog.Trace.Tests.TestHandler5::GenericBaseMethod3", "Datadog.Trace.Tests.GenericBaseHandler2`2", "GenericBaseMethod3", ClrNames.Void, "!1")]
+        [InlineData("Datadog.Trace.Tests::Datadog.Trace.Tests.TestHandler5::GenericBaseMethod4", "Datadog.Trace.Tests.GenericBaseHandler2`2", "GenericBaseMethod4", "!1")]
+        [InlineData("Datadog.Trace.Tests::Datadog.Trace.Tests.TestHandler5::GenericBaseMethod5", "Datadog.Trace.Tests.GenericBaseHandler2`2", "GenericBaseMethod5", "!1", "!0")]
+        [InlineData("Datadog.Trace.Tests::Datadog.Trace.Tests.TestHandler5::GenericBaseMethod6", "Datadog.Trace.Tests.GenericBaseHandler2`2", "GenericBaseMethod6", ClrNames.Void, "!0", "!1")]
+        [InlineData("Datadog.Trace.Tests::Datadog.Trace.Tests.TestHandler5::GenericBaseMethod7", "Datadog.Trace.Tests.GenericBaseHandler2`2", "GenericBaseMethod7", "System.Threading.Tasks.Task`1[!0]", "System.Threading.Tasks.Task`1[!1]")]
+        [InlineData("Datadog.Trace.Tests::Datadog.Trace.Tests.TestHandler6::GenericBaseMethod5", "Datadog.Trace.Tests.GenericBaseHandler2`2", "GenericBaseMethod5", "!1", "!0")]
+        [InlineData("Datadog.Trace.Tests::Datadog.Trace.Tests.TestHandler6::GenericBaseMethod6", "Datadog.Trace.Tests.GenericBaseHandler2`2", "GenericBaseMethod6", ClrNames.Void, "!0", "!1")]
+        [InlineData("Datadog.Trace.Tests::Datadog.Trace.Tests.TestHandler6::GenericBaseMethod7", "Datadog.Trace.Tests.GenericBaseHandler2`2", "GenericBaseMethod7", "System.Threading.Tasks.Task`1[!0]", "System.Threading.Tasks.Task`1[!1]")]
+        [InlineData("Datadog.Trace.Tests::Datadog.Trace.Tests.TestHandler7+Nested::Handler", "Datadog.Trace.Tests.TrickyParamHandler+NestedGeneric`2", "Handler", "System.Threading.Tasks.Task`1[System.Tuple`2[!0,NestedGeneric`2[!0,!1]]]", "System.Tuple`2[!0,NestedGeneric`2[!0,!1]]")]
+        [InlineData("Datadog.Trace.Tests::Datadog.Trace.Tests.FunctionImplementation::FunctionHandlerAsync", "Datadog.Trace.Tests.AbstractAspNetCoreFunction`2", "FunctionHandlerAsync", "System.Threading.Tasks.Task`1[!1]", "!0", "Datadog.Trace.Tests.ILambdaContext")]
+        [InlineData("Datadog.Trace.Tests::Datadog.Trace.Tests.DerivedImplementation::ComplexNestedGeneric", "Datadog.Trace.Tests.GenericBase`2", "ComplexNestedGeneric", "Datadog.Trace.Tests.GenericBase`2[Datadog.Trace.Tests.CustomInput,InnerGeneric`2[!0,!1]]", "!0", "InnerGeneric`2[!0,NestedInSameType`2[!0,!1,!0,System.Collections.Generic.Dictionary`2[System.String,DeepNested`1[!0,!1,!1]]]]")]
+        [InlineData("Datadog.Trace.Tests::Datadog.Trace.Tests.DerivedImplementation+NestedDerived::ComplexNestedGeneric", "Datadog.Trace.Tests.GenericBase`2", "ComplexNestedGeneric", "Datadog.Trace.Tests.GenericBase`2[Datadog.Trace.Tests.CustomInput,InnerGeneric`2[!0,!1]]", "!0", "InnerGeneric`2[!0,NestedInSameType`2[!0,!1,!0,System.Collections.Generic.Dictionary`2[System.String,DeepNested`1[!0,!1,!1]]]]")]
```

Comment:
I don't understand the the following:

- `NestedInSameType`2[!0,!1,!0,System.Collections.Generic.Dictionary`2[...]`
- `System.Collections.Generic.Dictionary`2[System.STring,DeepNested`1[!0,!1,!1]]`

It looks like there are unnecessary generic type variables in each of these. Is this expected?

Reply:
Believe me, I don't fully understand it either 😂 But it happens in the scenario where you have a generic class (with `x` generic params) nested _inside_ another generic class (with `y` generic params). In that case, you need to provide all `x+y` args to fully define the type. 

The good news is that I know this construction us correct because these unit test examples are taken almost directly from the integration tests, so I can confirm that the profiler hooks into it as expected:

```bash
Looking for 'Samples.AWS.Lambda.GenericBase`2.ComplexNestedGeneric(2 params)' method implementation.
* Comparing signature for method: Samples.AWS.Lambda.GenericBase`2.ComplexNestedGeneric
    -> InnerGeneric`2[!0,NestedInSameType`2[!0,!1,!0,System.Collections.Generic.Dictionary`2[System.String,DeepNested`1[!0,!1,!1]]]] 
     = InnerGeneric`2[!0,NestedInSameType`2[!0,!1,!0,System.Collections.Generic.Dictionary`2[System.String,DeepNested`1[!0,!1,!1]]]]
    -> Amazon.Lambda.Core.ILambdaContext
     = Amazon.Lambda.Core.ILambdaContext
```

For that example we have the following (ridiculously convoluted) class hierachy:

```csharp
public class DerivedImplementation : GenericBase<CustomInput, CustomStruct>
{
    public class NestedDerived : GenericBase<CustomInput, CustomStruct>
    {
    }
}

public abstract class GenericBase<TRequest, TResponse>
{
    public GenericBase<CustomInput, Function.NestedClass.InnerGeneric<TRequest, TResponse>> ComplexNestedGeneric(
            Function.NestedClass.InnerGeneric<TRequest, NestedInSameType<TRequest, Dictionary<string, Nested.DeepNested<TResponse>>>> request,
            ILambdaContext context)
        {
            HandleRequest((TRequest)default);
            return default;
        }
    
    public class NestedInSameType<TKey, TValue> : Dictionary<TKey, TValue>
    {
    }

    public class Nested
    {
        public class DeepNested<TNested> : HashSet<TNested>
        {
        }
    }
}
```

And it works 😄

Reply:
Wow, this is quite complicated indeed, but it makes sense to me. Thanks for the explanation!